### PR TITLE
[PRW 2.0]  Moved rw2 proto to the full path (both package name and placement)

### PR DESCRIPTION
**Pull request by bwplotka**
Changes:
* New path: `github.com/prometheus/prometheus/prompb/io/prometheus/write/v2`, so `io/prometheus/write/v2` package. This is what buf recommends https://buf.build/docs/reference/protobuf-files-and-packages#packages and i shortened it (we will likely never have "local write").
* WriteRequest -> Request.
* Updated commentary + grammarly.
* Change behaviour for samples and histograms. It makes sense more I think, given metadata will be different for the known case this happens too.

### CODE REVIEW

**Comment by cstyan** [comment]
> Change behaviour for samples and histograms. It makes sense more I think, given metadata will be different for the known case this happens too.

Looks fine to me but I'm not seeing :point_up: this change, which I think you intended to commit (based on your note there but also the changes to the comments in the proto files).

File: prompb/io/prometheus/write/v2/types.proto [hunk 1]
```diff
@@ -11,58 +11,55 @@
...
-  // Samples are sorted by timestamp (older first).
+  // Samples and histograms are sorted by timestamp (older first).
   repeated Sample samples = 2 [(gogoproto.nullable) = false];
-  // histograms contain zero or more histogram samples for a given timeseries.
-  // For typical clients, in healthy cases, there will be only one sample, for ~real
-  // time metric streaming. histograms can co-exist with samples (see samples
-  // for details).
-  //
-  // histograms are sorted by timestamp (older first).
   repeated Histogram histograms = 3 [(gogoproto.nullable) = false];
```
Comment (bwplotka) [inline prompb/io/prometheus/write/v2/types.proto:60]: Changes to samples & histogram samples behaviour I mentioned.

**Comment by bwplotka** [comment]
Those changes are here @cstyan 🤗 https://github.com/prometheus/prometheus/pull/13973/files#r1592219265

File: prompb/io/prometheus/write/v2/types.proto (hunk 1 above)
Comment (bwplotka) [reply prompb/io/prometheus/write/v2/types.proto:60]: Relevant discussion https://docs.google.com/document/d/1PljkX3YLLT-4f7MqrLt7XCVPG3IsjRREzYrUzBxCPV0/edit?disco=AAAA3dp46Gw

**Comment by bwplotka** [comment]
Merging this, we can revert the commentary about samples vs histogram sample semantics later on, adding to our agenda

**Pull request merged**